### PR TITLE
Minor style fixes in deeplabv3

### PR DESCRIPTION
**Pull request by fchollet**

### CODE REVIEW

File: keras_cv/models/segmentation/deeplab.py [hunk 1]
```diff
@@ -89,22 +78,22 @@ def __init__(
...
-
-        feature_map = backbone(x)
+        feature_map = backbone(inputs)
         if spatial_pyramid_pooling is None:
             spatial_pyramid_pooling = SpatialPyramidPooling(
                 dilation_rates=[6, 12, 18]
             )
 
         output = spatial_pyramid_pooling(feature_map)
-        output = tf.keras.layers.UpSampling2D(
```
Comment (ianstenbit) [inline keras_cv/models/segmentation/deeplab.py:107]: We've made this mistake in a lot of places in KerasCV, so I opened #1527 to clean this up

File: keras_cv/models/segmentation/deeplab.py [hunk 2]
```diff
@@ -180,39 +177,42 @@ def get_config(self):
...
             no resize will happen to the output mask tensor.
-        kernel_size: default 3; the kernel_size to be used in each of the `convs` blocks
-        use_bias: default False; whether to use bias or not in each of the `convs` blocks
-                Defaults to none since the blocks use `BatchNormalization` after each conv, rendering
-                bias obsolete
-        activation: default 'softmax', the activation to apply in the classification
-            layer (output of the head)
-
-    Sample code
+        kernel_size: Int. The kernel_size to be used in each of the convolutional blocks.
```
Comment (ianstenbit) [inline keras_cv/models/segmentation/deeplab.py:206]: nit: lower-case `int`?

File: keras_cv/models/segmentation/deeplab.py [hunk 3]
```diff
@@ -180,39 +177,42 @@ def get_config(self):
...
+@keras.utils.register_keras_serializable(package="keras_cv")
 class SegmentationHead(layers.Layer):
     """Prediction head for the segmentation model
 
     The head will take the output from decoder (eg FPN or ASPP), and produce a
     segmentation mask (pixel level classifications) as the output for the model.
 
     Args:
-        num_classes: int, the number of output classes for the prediction. This should
+        num_classes: int. Number of output classes for the prediction. This should
```
Comment (ianstenbit) [inline keras_cv/models/segmentation/deeplab.py:192]: We've been using commas to separate the type from the description (e.g. in `output_scale_factor` below) -- should we be using periods instead throughout?

**Review by ianstenbit — COMMENTED** [review]
Thank you @fchollet! Just a few questions

I am thinking we should include a link to the [governance doc](https://github.com/keras-team/governance/blob/master/keras_api_design_guidelines.md) in our contributing guide to hopefully do better about these style inconsistencies moving forward. wdyt?

File: keras_cv/models/segmentation/deeplab.py (hunk 2 above)
Comment (fchollet) [reply keras_cv/models/segmentation/deeplab.py]: Done

File: keras_cv/models/segmentation/deeplab.py (hunk 3 above)
Comment (fchollet) [reply keras_cv/models/segmentation/deeplab.py]: No strong opinion as long as it's consistent (ideally, across all of our packages). I reverted the docstring to use commas.

**Review by ianstenbit — APPROVED** [review]
Thank you Francois!

**Comment by ianstenbit** [comment]
Failing GCB tests are due to the fact that this feature branch is branched from head before #1517, and they're unrelated to this PR.

**Pull request merged**